cmd/gcp-controller-manager/csr_approver.go

```diff
@@ -371,7 +371,7 @@ func validateNodeServerCert(ctx *controllerContext, csr *capi.CertificateSigning
 		for _, dns := range x509cr.DNSNames {
 			// Linux DNSName should be as the format of [INSTANCE_NAME].c.[PROJECT_ID].internal when using the global DNS, and [INSTANCE_NAME].[ZONE].c.[PROJECT_ID].internal when using zonal DNS.
 			// Windows DNSName should be INSTANCE_NAME
-			if dns != instanceName && dns != fmt.Sprintf("%s.c.%s.internal", instanceName, ctx.gcpCfg.ProjectID) && dns != fmt.Sprintf("%s.%s.c.%s.internal", instanceName, z, ctx.gcpCfg.ProjectID) {
+			if !strings.HasPrefix(dns, instanceName) {
```

Comment:
HasPrefix seems too weak here... that permits approving arbitrary `$instancename.$arbitrarydomain`, or even `$instancename$arbitrarysuffix.$arbitrarydomain`

Reply:
Format the projectID specific for the Domain-scoped project and added tests for it. Please take another look, thx!